pkgs/by-name/bi/bitfocus-companion/package.nix

```
@@ -0,0 +1,137 @@
+{
+  stdenv,
+  lib,
+  fetchFromGitHub,
+  nodejs,
+  git,
+  python3,
+  udev,
+  yarn-berry_4,
+  libusb1,
+  dart-sass,
+  electron_36,
+  makeWrapper,
+  nix-update-script,
+}:
+
+let
+  yarn-berry = yarn-berry_4;
+
+  selectSystem =
+    attrs:
+    attrs.${stdenv.hostPlatform.system} or (throw "Unsupported system: ${stdenv.hostPlatform.system}");
+  platform = selectSystem {
+    x86_64-linux = "linux-x64";
+    aarch64-linux = "linux-arm64";
+    armv7l-linux = "linux-armv7l";
+  };
+in
+
+stdenv.mkDerivation rec {
+  pname = "bitfocus-companion";
+  version = "4.0.3";
+
+  src = fetchFromGitHub {
+    owner = "bitfocus";
+    repo = "companion";
+    tag = "v${version}";
+    hash = "sha256-qe5PfiPgRuvua7FqM7rvmmPruHxbpHCR3iJ+NeXfr5Q=";
+  };
+
+  passthru.updateScript = nix-update-script { };
+
+  postPatch = ''
+    # patch out git calls to generate version strings.
+    substituteInPlace tools/lib.mts --replace-fail "return await fcn()" "return \"v${version}\""
+
+    # remove the yarn install during the build, since there is no internet connection, and everything has already been installed by yarnBerryConfigHook
+    substituteInPlace tools/build/dist.mts \
+      --replace-fail 'await $`yarn --cwd node_modules/better-sqlite3 prebuild-install --arch=''${platformInfo.nodeArch}`' ""
+
+    substituteInPlace tools/build/package.mts --replace-fail "await $\`yarn install --no-immutable\`" ""
+
+    # remove node download, since we'll use the nix version
+    substituteInPlace tools/build/package.mts --replace-fail "const nodeVersions = await fetchNodejs(platformInfo)" "const nodeVersions = []"
+    substituteInPlace tools/build/package.mts --replace-fail "await fs.createSymlink(latestRuntimeDir, path.join(runtimesDir, 'main'))" ""
+  '';
+
+  nativeBuildInputs = [
+    nodejs
+    yarn-berry.yarnBerryConfigHook
+    git
+    python3
+    udev
+    electron_36
+    yarn-berry
+    makeWrapper
+  ];
+
+  buildInputs = [
+    libusb1
+    dart-sass
+    nodejs
+  ];
+
+  missingHashes = ./missing-hashes.json;
+
+  offlineCache = yarn-berry.fetchYarnBerryDeps {
+    inherit src missingHashes;
+    hash = "sha256-6X6AEhgKDcZT8whQV00t6BdXSOJH9ZtE4YtJ/vKhgbI=";
+  };
+
+  env = {
+    ELECTRON_SKIP_BINARY_DOWNLOAD = 1;
+    SKIP_LAUNCH_CHECK = true;
+    ELECTRON = 0;
+  };
+
+  configurePhase = ''
+    runHook preConfigure
+    runHook postConfigure
+  '';
+
+  buildPhase = ''
+    runHook preBuild
+
+    # force sass-embedded to use our own sass instead of the bundled one
+    substituteInPlace node_modules/sass-embedded/dist/lib/src/compiler-path.js \
+        --replace-fail 'compilerCommand = (() => {' 'compilerCommand = (() => { return ["${lib.getExe dart-sass}"];'
+
+    yarn lindist ${platform}
```

The platform argument here is ignored.

```suggestion
    yarn dist ${platform}
```